### PR TITLE
V0.12

### PR DESCRIPTION
**Pull request by aluchianic**
## Type of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] This change requires a change to the documentation

## How Has This Been Tested?

- [x] [root] terraform init, plan, apply
- [x] [example_reader] terraform init, plan, apply
- [x] [example_reader_config] terraform init, plan

## Checklist:

- [x] I have read the **CONTRIBUTING** document.
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have checked that my changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: components/landing_zone_cloudtrail/.terrahub.yml [hunk 1]
```diff
@@ -20,26 +20,26 @@ component:
         landing_zone_cloudtrail:
           provider: 'aws.${tfvar.terrahub["landing_zone_providers"]["0"]}'
           s3_bucket_name: >-
-            ${data.terraform_remote_state.landing_zone_cloudtrail_s3_bucket.bucket}
-          name: '${var.landing_zone_cloudtrail_name}'
-          is_multi_region_trail: '${var.landing_zone_cloudtrail_is_multi_region_trail}'
-          is_organization_trail: '${var.landing_zone_cloudtrail_is_organization_trail}'
+            data.terraform_remote_state.landing_zone_cloudtrail_s3_bucket.bucketZ
```
Comment (eistrati) [inline components/landing_zone_cloudtrail/.terrahub.yml]: `bucketZ`?

File: .terrahub.yml [hunk 2]
```diff
@@ -14,15 +14,24 @@ terraform:
...
+        assume_role:
+          - role_arn: >-
+              arn:aws:iam::var.sample_account_id:role/OrganizationAccountAccessRole
+            session_name: var.sample_account_id
   tfvars:
-    account_id: 123456789012
+    account_id: 914196360175
     region: us-east-1
     tfstate_path: /tmp/.terrahub/tfstate_local/terraform-aws-landing-zone
+    sample_account_id: 123456789015
```
Comment (eistrati) [inline .terrahub.yml]: rename `123456789015` to `987654321098`

File: .terrahub.yml [hunk 3]
```diff
@@ -14,15 +14,24 @@ terraform:
...
+    - aws:
+        alias: sample
+        region: var.sample_region
+        assume_role:
+          - role_arn: >-
+              arn:aws:iam::var.sample_account_id:role/OrganizationAccountAccessRole
+            session_name: var.sample_account_id
   tfvars:
-    account_id: 123456789012
+    account_id: 914196360175
```
Comment (eistrati) [inline .terrahub.yml]: revert back to `123456789012`

File: examples/example_landing_zone_reader/README.md [hunk 4]
```diff
@@ -7,15 +7,14 @@ This module can be used as easy as:
 ```hcl
 module "example_landing_zone_reader" {
   source  = "../../modules/landing_zone_reader"
-  version = "0.0.8"
```
Comment (eistrati) [inline examples/example_landing_zone_reader/README.md:10]: please revert back and don't remove version from examples

File: examples/example_landing_zone_reader/main.tf [hunk 5]
```diff
@@ -14,8 +13,8 @@ resource "aws_lambda_function" "hello_world" {
   s3_bucket     = "www.terrahub.io"
   s3_key        = "/hello-world/nodejs10.x.zip"
 
-  vpc_config = {
-    subnet_ids         = ["${local.landing_zone_subnet_ids["default"]}"]
-    security_group_ids = ["${local.landing_zone_security_group_ids["default"]}"]
+  vpc_config {
+    security_group_ids = local.landing_zone_subnet_ids["default"]
```
Comment (eistrati) [inline examples/example_landing_zone_reader/main.tf]: why `security_group_ids` are inheriting values from landing_zone_subnet_ids`?

File: examples/example_landing_zone_reader_config/README.md [hunk 6]
```diff
@@ -6,13 +6,12 @@ This example implements fully functional terraform configuration that can progra
...
 ```hcl
 module "example_landing_zone_reader_config" {
-  source    = "../../modules/landing_zone_reader_config"
-  version   = "0.0.8"
-  root_path = "${path.cwd}/../.."
-
-  landing_zone_providers  = "${var.landing_zone_providers}"
-  landing_zone_components = "${var.landing_zone_components}"
+  source                  = "../../modules/landing_zone_reader_config"
+  root_path               = "${path.cwd}/../.."
```
Comment (eistrati) [inline examples/example_landing_zone_reader_config/README.md:10]: revert back `version`

File: examples/example_landing_zone_s3_and_codepipeline/README.md [hunk 7]
```diff
@@ -5,10 +5,9 @@ This example implements fully functional terraform configuration that can progra
 
 This module can be used as easy as:
 ```hcl
-module "example_landing_zone_organization" {
+module "example_landing_zone_s3_and_codepipeline" {
   source    = "../../modules/landing_zone"
-  version   = "0.0.8"
```
Comment (eistrati) [inline examples/example_landing_zone_s3_and_codepipeline/README.md:10]: revert back `version`

File: examples/example_landing_zone_s3_and_codepipeline/main.tf [hunk 8]
```diff
@@ -1,7 +1,7 @@
 module "example_landing_zone_s3_and_codepipeline" {
   source    = "../../modules/landing_zone"
-  version   = "0.0.8"
```
Comment (eistrati) [inline examples/example_landing_zone_s3_and_codepipeline/main.tf:3]: revert back `version`

File: main.tf [hunk 9]
```diff
@@ -1,7 +1,7 @@
 module "landing_zone" {
   source                  = "./modules/landing_zone"
-  version                 = "0.0.8"
```
Comment (eistrati) [inline main.tf:3]: revert back `version`

File: modules/landing_zone/scripts/destroy.js [hunk 10]
```diff
@@ -34,6 +34,8 @@ async function main() {
   try {
     await Helper.executeWithErrors('terrahub', processes, rootPath);
   } catch (error) {
+    console.log(`ERR : `, error);
```
Comment (eistrati) [inline modules/landing_zone/scripts/destroy.js]: rename `ERR  :` to  `ERROR:`

File: modules/landing_zone/scripts/helper.js [hunk 11]
```diff
@@ -124,11 +240,7 @@ class Helper {
 
       subKeys.forEach(subKey => {
         if (json[key][subKey]['value']) {
-          if (compressing) {
-            response[subKey] = this.getOutputValueByType(json[key][subKey]['value']);
-          } else {
             response[subKey] = json[key][subKey]['value'];
```
Comment (eistrati) [inline modules/landing_zone/scripts/helper.js]: fix indentation

File: modules/landing_zone_reader_config/scripts/apply.js [hunk 12]
```diff
@@ -41,11 +41,22 @@ async function terrahubOutput(include) {
...
+      );
+
       const result = await Helper.cli(
         'terrahub',
         ['output', '--format', 'json', '--include', item, '--auto-approve'],
         rootPath
       );
+
+      if (!result.length) {
+        throw new Error(`First you need to run LandingZone module, because you don't have resources in AWS`);
```
Comment (eistrati) [inline modules/landing_zone_reader_config/scripts/apply.js]: please rename error message to ```No terraform outputs found. Before using `landing_zone_reader` module, make sure that `landing_zone` module generates output. Learn more: https://github.com/TerraHubCorp/terraform-aws-landing-zone/```

File: terraform.tfvars [hunk 13]
```diff
@@ -1,46 +1,45 @@
 landing_zone_providers = {
   default = {
-    account_id = "123456789012"
-    region     = "us-east-1"
+    account_id = "914196360175"
```
Comment (eistrati) [inline terraform.tfvars]: revert back account_id

File: terraform.tfvars [hunk 14]
```diff
@@ -1,46 +1,45 @@
...
+    region = "us-east-1"
   }
 
-  # ,
-  # sample = {
-  #   account_id = "123456789012"
-  #   region     = "us-east-1"
-  # }
+//  sample = {
```
Comment (eistrati) [inline terraform.tfvars]: revert back `#` and change account_id to `987654321098`

File: terraform.tfvars [hunk 15]
```diff
@@ -1,46 +1,45 @@
...
-  landing_zone_security_group                 = "s3://terraform-aws-landing-zone/mycompany/landing_zone_security_group/config.tfvars"
-  landing_zone_security_group_rule_ingress    = "s3://terraform-aws-landing-zone/mycompany/landing_zone_security_group_rule_ingress/config.tfvars"
-  landing_zone_security_group_rule_egress     = "s3://terraform-aws-landing-zone/mycompany/landing_zone_security_group_rule_egress/config.tfvars"
-  landing_zone_cloudtrail_s3_bucket           = "s3://terraform-aws-landing-zone/mycompany/landing_zone_cloudtrail_s3_bucket/config.tfvars"
-  landing_zone_cloudtrail_s3_bucket_policy    = "s3://terraform-aws-landing-zone/mycompany/landing_zone_cloudtrail_s3_bucket_policy/config.tfvars"
-  landing_zone_cloudtrail                     = "s3://terraform-aws-landing-zone/mycompany/landing_zone_cloudtrail/config.tfvars"
-  landing_zone_directory_service_directory    = "s3://terraform-aws-landing-zone/mycompany/landing_zone_directory_service_directory/config.tfvars"
+    landing_zone_pipeline_s3_bucket             = "s3://terraform-aws-landing-zone/mycompany/landing_zone_pipeline_s3_bucket/config.tfvars"
```
Comment (eistrati) [inline terraform.tfvars]: revert back indentation

**Review by eistrati — CHANGES_REQUESTED** [review]

File: components/landing_zone_cloudtrail/.terrahub.yml (hunk 1 above)
Comment (aluchianic) [reply components/landing_zone_cloudtrail/.terrahub.yml]: updated

File: .terrahub.yml (hunk 2 above)
Comment (aluchianic) [reply .terrahub.yml]: updated

File: .terrahub.yml (hunk 3 above)
Comment (aluchianic) [reply .terrahub.yml]: updated

File: modules/landing_zone_reader_config/scripts/apply.js (hunk 12 above)
Comment (aluchianic) [reply modules/landing_zone_reader_config/scripts/apply.js]: updated

File: terraform.tfvars (hunk 14 above)
Comment (aluchianic) [reply terraform.tfvars]: updated

File: terraform.tfvars (hunk 13 above)
Comment (aluchianic) [reply terraform.tfvars]: updated

File: modules/landing_zone/scripts/helper.js (hunk 11 above)
Comment (aluchianic) [reply modules/landing_zone/scripts/helper.js]: fixed

File: examples/example_landing_zone_reader/main.tf (hunk 5 above)
Comment (aluchianic) [reply examples/example_landing_zone_reader/main.tf]: fixed

File: examples/example_landing_zone_organization/README.md [hunk 16]
```diff
@@ -8,7 +8,7 @@ This module can be used as easy as:
 module "example_landing_zone_organization" {
   source    = "../../modules/landing_zone"
   version   = "0.0.8"
```
Comment (eistrati) [inline examples/example_landing_zone_organization/README.md:10]: remove `version`

File: examples/example_landing_zone_organization/main.tf [hunk 17]
```diff
@@ -1,7 +1,8 @@
 module "example_landing_zone_organization" {
   source    = "../../modules/landing_zone"
   version   = "0.0.8"
```
Comment (eistrati) [inline examples/example_landing_zone_organization/main.tf]: remove `version`

File: examples/example_landing_zone_s3_and_codepipeline/main.tf [hunk 18]
```diff
@@ -1,7 +1,7 @@
...
-  root_path = "${path.module}"
+  root_path = path.module
+  output_path = pathexpand("~/.terrahub/output.json")
```
Comment (eistrati) [inline examples/example_landing_zone_s3_and_codepipeline/main.tf]: please remove `output_path` as input parameter. update the internal scripts to use `~/.terrahub/cache/output/[project_code]/output.json` that can be used across modules ;)

File: main.tf [hunk 19]
```diff
@@ -1,7 +1,7 @@
...
-  root_path               = "${path.module}"
-  landing_zone_providers  = "${var.landing_zone_providers}"
-  landing_zone_components = "${var.landing_zone_components}"
+  root_path               = path.module
+  output_path             = pathexpand("~/.terrahub/output.json")
```
Comment (eistrati) [inline main.tf]: please remove `output_path` as input parameter. update the internal scripts to use `~/.terrahub/cache/output/[project_code]/output.json` that can be used across modules ;)

File: modules/landing_zone/data.tf [hunk 20]
```diff
@@ -1,4 +1,4 @@
-data "external" "landing_zone_output_file" {
-  depends_on = ["null_resource.landing_zone_apply"]
-  program    = ["sh", "${path.module}/scripts/output.sh", "${var.root_path}/output.json"]
+data "local_file" "landing_zone_output_file" {
+  depends_on = [null_resource.landing_zone_apply]
+  filename   = var.output_path
```
Comment (eistrati) [inline modules/landing_zone/data.tf]: please remove `output_path` as variable parameter. update the internal scripts to use `~/.terrahub/cache/output/[project_code]/output.json` that can be used across modules ;)

File: modules/landing_zone/main.tf [hunk 21]
```diff
@@ -22,29 +22,30 @@ resource "null_resource" "landing_zone_config" {
...
 
   provisioner "local-exec" {
     when    = "create"
     command = "node ${path.module}/scripts/apply.js"
 
     environment = {
-      ROOT_PATH  = "${var.root_path}"
-      COMMAND    = "${var.landing_zone_command}"
-      COMPONENTS = "${jsonencode(var.landing_zone_components)}"
+      OUTPUT_PATH = var.output_path
```
Comment (eistrati) [inline modules/landing_zone/main.tf]: please remove `output_path` as environment variable. update the internal scripts to use `~/.terrahub/cache/output/[project_code]/output.json` that can be used across modules ;)

**Review by eistrati — CHANGES_REQUESTED** [review]

**Review by eistrati — APPROVED** [review]

File: .terrahub.yml [hunk 22]
```diff
@@ -14,15 +14,24 @@ terraform:
...
+        region: var.sample_region
+        assume_role:
+          - role_arn: >-
+              arn:aws:iam::var.sample_account_id:role/OrganizationAccountAccessRole
+            session_name: var.sample_account_id
   tfvars:
     account_id: 123456789012
     region: us-east-1
     tfstate_path: /tmp/.terrahub/tfstate_local/terraform-aws-landing-zone
+    sample_account_id: 987654321098
```
Comment (vcalmic) [inline .terrahub.yml:36]: Please change this account id into `123456789012`

File: terraform.tfvars [hunk 23]
```diff
@@ -3,11 +3,11 @@ landing_zone_providers = {
...
     region     = "us-east-1"
   }
-  # ,
-  # another_account = {
-  #   account_id = "987654321098"
-  #   region     = "us-east-1"
-  # }
+
+#  sample = {
+#    account_id = "123456789015"
```
Comment (vcalmic) [inline terraform.tfvars]:
```suggestion
#    account_id = "123456789012"
```

**Review by vcalmic — CHANGES_REQUESTED** [review]

File: .terrahub.yml (hunk 22 above)
Comment (eistrati) [reply .terrahub.yml:36]: ignore this. it is expected a different value, to show a different account. closing it now as resolved.

File: terraform.tfvars (hunk 23 above)
Comment (eistrati) [reply terraform.tfvars]: well, this one I've missed. please change it from `123456789012` to `987654321098`

**Review by vcalmic — APPROVED** [review]

**Pull request merged**